### PR TITLE
Cancel ongoing PoW if the tx was seen in another block

### PR DESCRIPTION
**Pull request by muXxer**

### CODE REVIEW

File: packages/nodeconn/nc_chain.go [hunk 1]
```diff
@@ -235,8 +235,8 @@ func (ncc *ncChain) getLastPendingTx() *pendingTransaction {
 	defer ncc.lastPendingTxLock.Unlock()
 
 	if ncc.lastPendingTx != nil {
-		ncc.nodeConn.pendingTransactionsLock.Lock()
-		defer ncc.nodeConn.pendingTransactionsLock.Unlock()
+		ncc.nodeConn.pendingTransactionsLock.RLock()
```
Comment (muXxer) [inline packages/nodeconn/nc_chain.go:238]: Damn, that change is wrong 😭
Can someone revert that? I'm not at my PC rn.

Comment (muXxer) [reply packages/nodeconn/nc_chain.go:238]: Ah, it was correct tho. 
`ncc.lastPendingTx` is secured by `lastPendingTxLock`, not `pendingTransactionsLock`, so this change is fine.

File: packages/nodeconn/nodeconn.go [hunk 2]
```diff
@@ -486,6 +503,83 @@ func (nc *nodeConnection) triggerChainCallbacks(ledgerUpdate *ledgerUpdate) erro
...
+
+			// block not referenced yet
+			// => check if the quality of the tips is good or if the block can never be referenced
+			if metadata.ShouldReattach {
+				// we can abort the block metadata check, but we should not abort our own attachment of the tx
+				break
+			}
+
+			// block is solid and the quality of the tips seem fine
+			// => abort out own attachment
```
Comment (jorgemmsilva) [inline packages/nodeconn/nodeconn.go]:
```suggestion
			// => abort our own attachment
```

File: packages/nodeconn/nodeconn.go [hunk 3]
```diff
@@ -486,6 +503,83 @@ func (nc *nodeConnection) triggerChainCallbacks(ledgerUpdate *ledgerUpdate) erro
...
+			cancelTxAttachment = true
+			break
+		}
+
+		if cancelTxAttachment {
+			pendingTx.SetPublished()
+		}
```
Comment (jorgemmsilva) [inline packages/nodeconn/nodeconn.go]: could just call `pendingTx.SetPublished()` in 2 places and remove the logic about `cancelTxAttachment`

**Review by jorgemmsilva — APPROVED** [review]

File: packages/nodeconn/nodeconn.go (hunk 3 above)
Comment (muXxer) [reply packages/nodeconn/nodeconn.go]: Changed! :+1:

**Pull request merged**